packages/core/src/schema/utils.ts

```diff
@@ -1,59 +1,73 @@
 import type {
+  Column,
+  Enum,
   EnumColumn,
   ManyColumn,
-  NonReferenceColumn,
   OneColumn,
   ReferenceColumn,
-} from "./types.js";
-
-/**
- * Returns the name of the table that was referenced by a reference column
- */
-export const referencedTableName = (references: unknown) =>
-  (references as string).split(".")[0];
-
-export const isOneColumn = (
-  column:
-    | EnumColumn
-    | NonReferenceColumn
-    | ReferenceColumn
-    | ManyColumn
-    | OneColumn,
-): column is OneColumn => column._type === "o";
-
-export const isBaseColumn = (
-  column:
-    | EnumColumn
-    | NonReferenceColumn
-    | ReferenceColumn
-    | ManyColumn
-    | OneColumn,
-): column is NonReferenceColumn | ReferenceColumn => column._type === "b";
-
-export const isManyColumn = (
-  column:
-    | EnumColumn
-    | NonReferenceColumn
-    | ReferenceColumn
-    | ManyColumn
-    | OneColumn,
-): column is ManyColumn => column._type === "m";
-
-export const isEnumColumn = (
-  column:
-    | EnumColumn
-    | NonReferenceColumn
-    | ReferenceColumn
-    | ManyColumn
-    | OneColumn,
-): column is EnumColumn => column._type === "e";
-
-export const isReferenceColumn = (
-  column:
-    | EnumColumn
-    | NonReferenceColumn
-    | ReferenceColumn
-    | ManyColumn
-    | OneColumn,
-): column is ReferenceColumn =>
-  column._type === "b" && column.references !== undefined;
+  ScalarColumn,
+  Schema,
+  Table,
+} from "./common.js";
+
+export const isScalarColumn = (column: Column): column is ScalarColumn =>
+  column[" type"] === "scalar";
+
+export const isReferenceColumn = (column: Column): column is ReferenceColumn =>
+  column[" type"] === "reference";
+
+export const isOneColumn = (column: Column): column is OneColumn =>
+  column[" type"] === "one";
+
+export const isManyColumn = (column: Column): column is ManyColumn =>
+  column[" type"] === "many";
+
+export const isEnumColumn = (column: Column): column is EnumColumn =>
+  column[" type"] === "enum";
+
+export const isOptionalColumn = (column: Column): boolean => {
+  if (isManyColumn(column) || isOneColumn(column)) return false;
+  return column[" optional"];
+};
+
+export const isListColumn = (column: Column): boolean => {
+  if (isManyColumn(column) || isOneColumn(column) || isReferenceColumn(column))
+    return false;
+  return column[" list"];
+};
+
+export const isTable = (tableOrEnum: Schema[string]): tableOrEnum is Table =>
+  Array.isArray(tableOrEnum) === false;
+
+export const isEnum = (tableOrEnum: Schema[string]): tableOrEnum is Enum =>
+  Array.isArray(tableOrEnum) === true;
+
+export const schemaToTables = (
```

Comment:
Maybe `getTables`, `getEnums`? In context, it will always be called with a `schema` object as the arg so it's kind of self-documenting.